trieste/models/gpflow/sampler.py

```diff
@@ -631,18 +663,19 @@ def __init__(
             raise NotImplementedError(
                 f"ResampleableRandomFourierFeatureFunctions only work with models that either"
                 f"support get_kernel, get_observation_noise and get_internal_data or support "
-                f"get_kernel, get_observation_noise and get_internal_data;"
+                f"get_kernel and get_inducing_variables;"
                 f"but received {model.__repr__()}."
             )
 
         self._kernel = model.get_kernel()
         self._n_components = n_components
         super().__init__(self._kernel, self._n_components, dtype=tf.float64)
 
-        if isinstance(model, SupportsGetInternalData):
-            dummy_X = model.get_internal_data().query_points[0:1, :]
-        else:
+        if isinstance(model, SupportsGetInducingVariables):
```

Comment:
I was running into an error with the SGPR wrapper, SGPR wrapper is both `FeatureDecompositionInducingPointModel` and `FeatureDecompositionInternalDataModel`, which with some methods here means that either query points or inducing points are used. Now, the the order of checking what kind of model we have was not consistent and for this reason I was getting errors, I made it consistent across the file now and `SupportsGetInducingVariables` is checked, which means that the model that satisfies both will be seen as inducing variables one `FeatureDecompositionInducingPointModel` - for SGPR wrapper that is a correct choice. @henrymoss and @uri-granta does this solution looks good?
This implicit default might be a fragile thing, do you have a better idea how to do it perhaps?